### PR TITLE
Add support for CommonJS with Rhino

### LINKED ISSUE #6: Add support for CommonJS with Rhino
In the current release I don't believe there is a way to include other js files.

Rhino 1.7R3 now suport CommonJS. It would be great to be able to use this feature in ES.

[1] - http://www.angrycoding.com/2011/09/mozilla-rhino-commonjs-module-support.html

### PR DESCRIPTION
**Pull request by dadoonet**
You can add your javascript modules under `config/scripts/modules/` and use them
in scripts:

For example, add `config/scripts/modules/pi.js`

``` javascript
exports.getPI = function(value) {
    return 3.14;
};
```

You can use it in a script:

``` javascript
var pi = require('pi');
ctx.pi = pi.getPI();
```

Closes #6.
Closes #7.

### CODE REVIEW

**Comment by chirag04** [comment]
What is the status of this PR?

**Comment by dadoonet** [comment]
@tlrx Ping. Any chance you could have a look at it?

File: src/main/java/org/elasticsearch/script/javascript/JavaScriptScriptEngineService.java [hunk 1]
```diff
@@ -41,258 +45,287 @@
...
+
+	private Global globalScope;
+
+	private File modulesFile;
+
+	@Inject
+	public JavaScriptScriptEngineService(Settings settings, Environment env) {
+		super(settings);
+
+		this.optimizationLevel = componentSettings.getAsInt("optimization_level", 1);
```
Comment (wezhang) [inline src/main/java/org/elasticsearch/script/javascript/JavaScriptScriptEngineService.java]: Looks like the different indent space introduced lots of differences. Could they be cleaned to see the main changes? Just my 2 cents. :smile:

**Comment by dsernst** [comment]
Is this supported now? Unclear from documentation.

**Pull request closed without merging**